src/seaport.ts

```diff
@@ -649,6 +679,104 @@ export class OpenSeaPort {
     );
   }
 
+  private getAmountWithBasisPointsApplied = (
+    amount: BigNumber,
+    basisPoints: number
+  ) => {
+    return amount
+      .multipliedBy(basisPoints)
+      .dividedBy(INVERSE_BASIS_POINT)
+      .toString();
+  };
+
+  private async getFees({
+    openseaAsset: asset,
+    paymentTokenAddress,
+    startAmount,
+    endAmount,
+  }: {
+    openseaAsset: OpenSeaAsset;
+    paymentTokenAddress: string;
+    startAmount: BigNumber;
+    endAmount?: BigNumber;
+  }): Promise<{
+    sellerFee: ConsiderationInputItem;
+    openseaSellerFee: ConsiderationInputItem;
+    collectionSellerFee?: ConsiderationInputItem;
+    openseaBuyerFee?: ConsiderationInputItem;
+    collectionBuyerFee?: ConsiderationInputItem;
+  }> {
+    // Seller fee basis points
+    const openseaSellerFeeBasisPoints = DEFAULT_SELLER_FEE_BASIS_POINTS;
+    const collectionSellerFeeBasisPoints =
+      asset.collection.devSellerFeeBasisPoints;
+
+    // Buyer fee basis points
+    const openseaBuyerFeeBasisPoints = DEFAULT_BUYER_FEE_BASIS_POINTS;
+    const collectionBuyerFeeBasisPoints =
+      asset.collection.devBuyerFeeBasisPoints;
+
+    // Seller basis points
+    const sellerBasisPoints =
+      INVERSE_BASIS_POINT -
+      openseaSellerFeeBasisPoints -
+      collectionSellerFeeBasisPoints;
+
+    const getConsiderationItem = (basisPoints: number, recipient?: string) => {
+      return {
+        token: paymentTokenAddress,
+        amount: this.getAmountWithBasisPointsApplied(startAmount, basisPoints),
+        endAmount: this.getAmountWithBasisPointsApplied(
+          endAmount ?? startAmount,
+          basisPoints
+        ),
+        recipient,
+      };
+    };
+
+    return {
+      sellerFee: getConsiderationItem(sellerBasisPoints),
+      openseaSellerFee: getConsiderationItem(
+        openseaSellerFeeBasisPoints,
+        OPENSEA_FEE_RECIPIENT
+      ),
+      collectionSellerFee:
+        collectionSellerFeeBasisPoints > 0 && asset.collection.payoutAddress
+          ? getConsiderationItem(
+              collectionSellerFeeBasisPoints,
+              asset.collection.payoutAddress
+            )
+          : undefined,
+      openseaBuyerFee:
+        openseaBuyerFeeBasisPoints > 0
+          ? getConsiderationItem(
+              openseaBuyerFeeBasisPoints,
+              OPENSEA_FEE_RECIPIENT
+            )
+          : undefined,
+      collectionBuyerFee:
+        collectionBuyerFeeBasisPoints > 0 && asset.collection.payoutAddress
+          ? getConsiderationItem(
+              collectionBuyerFeeBasisPoints,
+              asset.collection.payoutAddress
+            )
+          : undefined,
+    };
+  }
+
+  private getAssetItems(
+    assets: Asset[],
+    quantities: number[] = [],
```

Comment:
Should we pass these quantities in as bignumber? number isn't reliable for non-int quantities.

Applies everywhere we're passing around amounts

Reply:
good point. I want the sdk consumer to be able to use a simple number when calling. I can make it take `BigNumberInput` and immediately convert to bignumber for internal use